src/Core/Batch/BatchDaemon.php

```diff
@@ -0,0 +1,203 @@
+                break;
+        }
+    }
+
+    /**
+     * A loop for the parent.
+     *
+     * @return void
+     */
+    public function runParent()
+    {
+        $procs = [];
+        while (true) {
+            $jobs = $this->runner->getJobs();
+            foreach ($jobs as $job) {
+                if (! array_key_exists($job->getIdentifier(), $procs)) {
+                    echo 'Spawning children' . PHP_EOL;
+                    $procs[$job->getIdentifier()] = [];
+                    for ($i = 0; $i < $job->getWorkerNum(); $i++) {
+                        $procs[$job->getIdentifier()][] = proc_open(
+                            sprintf('%s %d', $this->command, $job->getIdNum()),
+                            $this->descriptorSpec,
+                            $pipes
+                        );
+                    }
+                }
+            }
+            pcntl_signal_dispatch();
+            if ($this->shutdown) {
+                echo 'Shutting down, waiting for the children' . PHP_EOL;
+                foreach ($procs as $k => $v) {
+                    foreach ($v as $proc) {
+                        $status = proc_get_status($proc);
+                        // Keep sending SIGTERM until the child exits.
+                        while ($status['running'] === true) {
+                            @proc_terminate($proc);
+                            usleep(50000);
+                            $status = proc_get_status($proc);
+                        }
+                        @proc_close($proc);
+                    }
+                }
+                echo 'BatchDaemon exiting' . PHP_EOL;
+                exit;
+            }
+            usleep(1000000); // Reload the config after 1 second
+            // Reload the config
+            $this->runner->loadConfig();
+        }
+    }
+
+    /**
+     * A loop for the children.
+     *
+     * @param int $idNum Numeric id for the job.
+     * @return void
+     */
+    public function runChild($idNum)
+    {
+        // child process
+        $sysvKey = $this->getSysvKey($idNum);
+        $q = msg_get_queue($sysvKey);
+        $items = [];
+        $job = $this->runner->getJobFromIdNum($idNum);
+        $period = $job->getCallPeriod();
+        $lastInvoked = microtime(true);
+        $batchSize = $job->getBatchSize();
+        while (true) {
+            // Fire SIGALRM after 1 second to unblock the blocking call.
+            pcntl_alarm(1);
+            if (msg_receive(
+                $q,
+                0,
+                $type,
+                8192,
+                $message,
+                true,
+                0, // blocking mode
+                $errorcode
+            )) {
+                if ($type === self::$typeDirect) {
+                    $items[] = $message;
+                } elseif ($type === self::$typeFile) {
+                    $items[] = unserialize(file_get_contents($message));
+                    @unlink($message);
```

Comment:
I don't want the daemon to stop in any case of failure on this operation. The file will stay forever, but not a big problem, compared to a possibility of that the daemon stops.